### ISSUE TITLE
Drag and dropping .fbx assets from the Asset Browser has inconsistent behavior and may result in an assert

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#7426]
**Describe the bug**
Drag and dropping .fbx assets in different ways can result in a few different outcomes depending on where they are dropped. This was tested with dropping the asset:
- Into the Viewport
- Into the Entity Outliner
- (Linux issue: https://github.com/o3de/o3de/issues/7235) Into the Entity Outliner with no entity behind it
- (Linux issue: https://github.com/o3de/o3de/issues/7235) Into the Entity Outliner with an entity behind it

Different results observed were:
- No issue after dropping (An entity with Mesh component is created, and the proper Mesh asset is assigned)
- An entity with two mesh components and a wrong Mesh assigned is created
- Two entities are created, where one has tiny/large/regular scale, and wrong mesh assigned; and the other has the same mesh asset assigned but does not render
- One entity with the mesh is created but its scale is tiny/large
- After each of the different results an assert EntityOutlinerListModel.cpp(155) can be printed to the Console

Assert information:
File:
```
/home/gpc-610/o3de/Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutlinerListModel.cpp
```
Condition:
```
(139796455045376) 'virtual QModelIndex AzToolsFramework::EntityOutlinerListModel::index(int, int, const QModelIndex &) const'
No child found for parent
```
Line: 155

**Steps to reproduce**
1. In the Asset Browser find the Cube.fbx asset.
2. Drag and drop the asset into the Viewport and into the Entity Outliner.
3. (If on Linux with https://github.com/o3de/o3de/issues/7235 present): Set the camera so that half of the Entity Ouliner has no entity behind it, and the other has some entity behind it (e.g. Ground entity from the Atom Default Environment).
4. (If on Linux with https://github.com/o3de/o3de/issues/7235 present): Drag and drop the asset into both halves of the Entity Outliner (one with no entities behind it and one with an entity behind it).

**Expected behavior**
Mesh entities are properly and consistently instantiated regardless of where they are drag and dropped from the Asset Browser.

**Actual behavior**
Drag and dropping the .fbx assets from the Asset Browser has inconsistent results.

**Videos and Editor.log**
Videos of two different PCs on different Operating Systems:
Linux:

https://user-images.githubusercontent.com/86952082/152528643-9b59230f-835d-49d3-8182-12f7b8d9643c.mp4

Windows:

https://user-images.githubusercontent.com/86952082/152528676-b510aa68-3856-4512-8785-b407bbe1afbc.mp4

[Editor.log](https://github.com/o3de/o3de/files/8002898/Editor.log)

**Found in Branch**
Development (611598c)

**Desktop**
 - Device: PC
 - OS: Linux
 - Version: Ubuntu 20.04.3 LTS
 - CPU AMD Ryzen 5 3600
 - GPU NVIDIA GeForce RTX 2060 SUPER
 - Memory 16GB

**Comment by monroegm** [comment]
We will need guidance from @o3de/sig-ui-ux here

**Comment by alexmontAmazon** [comment]
@AMZN-stankowi I know you are already looking at the consistency of drag and dropping for fbx files. Will your work handle any of the above scenarios?

**Comment by AMZN-stankowi** [comment]
Yeah, this is what I'm working on.

**Comment by AMZN-stankowi** [comment]
Drag and drop behavior is now the same for files from the asset browser to the 3D viewport or outliner https://github.com/o3de/o3de/pull/7858

**Comment by AMZN-stankowi** [comment]
Drag and drop behavior has been updated a few times since this ticket was created, it should be fixed now.

**Comment by LB-ArturZieba** [comment]
Issue verified as fixed on the Development (14076b4) build.
